### PR TITLE
Add test cases for INTMIN/-1

### LINKED ISSUE #677: Add test cases to see how verifier handles -INT64_MIN
INT64_MIN/(-1) cannot be represented, nor can INT64_MIN*(-1), nor can -INT64_MIN.
We should probably add these as test cases.

### PR DESCRIPTION
**Pull request by dthaler**
Updates bpf_conformance to latest and adds the new test cases to the test pass

Fixes #677

## Summary by CodeRabbit

- **New Features**
	- Introduced several new conformance test cases to improve coverage of edge cases in integer operations, including multiplication, negation, division, and modulo.

- **Chores**
	- Updated the subproject commit reference for better alignment with the latest developments.

### CODE REVIEW

File: src/test/test_conformance.cpp [hunk 1]
```diff
@@ -219,11 +227,17 @@ TEST_CONFORMANCE("rsh64-reg-neg.data")
...
+TEST_CONFORMANCE("sdiv32-intmin-by-negone-reg.data")
 TEST_CONFORMANCE("sdiv32-reg.data")
 TEST_CONFORMANCE("sdiv64-by-zero-imm.data")
 TEST_CONFORMANCE("sdiv64-by-zero-reg.data")
 TEST_CONFORMANCE("sdiv64-imm.data")
+TEST_CONFORMANCE("sdiv64-intmin-by-negone-imm.data")
+TEST_CONFORMANCE("sdiv64-intmin-by-negone-reg.data")
 TEST_CONFORMANCE("sdiv64-reg.data")
+TEST_CONFORMANCE("smod32-intmin-by-negone-imm.data")
+TEST_CONFORMANCE("smod32-intmin-by-negone-reg.data")
```
[the remark this answers is not in the document]
Comment (dthaler) [reply src/test/test_conformance.cpp:240]: cases are listed in alphabetical order

File: src/test/test_conformance.cpp [hunk 2]
```diff
@@ -197,12 +197,20 @@ TEST_CONFORMANCE("movsx3264-reg.data")
...
+TEST_CONFORMANCE("mul32-intmin-by-negone-reg.data")
 TEST_CONFORMANCE("mul32-reg-overflow.data")
 TEST_CONFORMANCE("mul32-reg.data")
 TEST_CONFORMANCE("mul64-imm.data")
+TEST_CONFORMANCE("mul64-intmin-by-negone-imm.data")
+TEST_CONFORMANCE("mul64-intmin-by-negone-reg.data")
 TEST_CONFORMANCE("mul64-reg.data")
 TEST_CONFORMANCE("neg.data")
+TEST_CONFORMANCE("neg32-intmin-imm.data")
+TEST_CONFORMANCE("neg32-intmin-reg.data")
```
[the remark this answers is not in the document]
Comment (dthaler) [reply src/test/test_conformance.cpp:210]: There is a test case for every file in bpf_conformance.

File: src/test/test_conformance.cpp [hunk 3]
```diff
@@ -219,11 +227,17 @@ TEST_CONFORMANCE("rsh64-reg-neg.data")
 TEST_CONFORMANCE("sdiv32-by-zero-imm.data")
 TEST_CONFORMANCE("sdiv32-by-zero-reg.data")
 TEST_CONFORMANCE("sdiv32-imm.data")
+TEST_CONFORMANCE("sdiv32-intmin-by-negone-imm.data")
+TEST_CONFORMANCE("sdiv32-intmin-by-negone-reg.data")
```
[the remark this answers is not in the document]
Comment (dthaler) [reply src/test/test_conformance.cpp:231]: Does not seem important and if it were, they would be added in the bpf_conformance repository where these .data files are maintained rather than here.
